### ISSUE TITLE
gpx bb download before "start mapping"

### ISSUE THREAD

**Issue by hackery** [hotosm/osm-tasking-manager2#902]
Could the `${task_gpx_link}` be made available in the task template prior to hitting "Start Mapping"? That seems useful to me, to make a quick review of an area to consider, which otherwise would pollute the comment history with unnecessary lock/unlock messages.

**Comment by althio** [comment]
But we don't want people to start mapping without locking, so I am not sure this is the good solution to the problem. There is the risk of people bypassing the locking mechanism too easily.

I understand the problem of too much unnecessary lock/unlock messages in comment history, this was almost covered by this: https://github.com/hotosm/osm-tasking-manager2/pull/542 (following discussions https://github.com/hotosm/osm-tasking-manager2/issues/452 and related issues)

I agree also that sometimes you want to make a quick review (as a validator for exemple) while the task is already locked, but I don't see much value for a regular mapper.

**Comment by bgirardot** [comment]
I agree @althio Simplifying the start-stop process is a major goal, that is why it is a tag, it is at the root of a lot of dicussions.

This gave me the idea for an "Advanced" panel somewhere or setting, so if you set yourself advanced you see some other options, but you have to go find the preference and switch it.

**Comment by hackery** [comment]
@althio yes, I see that could be a risk. An option that's not initially shown might be preferable, or yes, in an "Advanced" or "Expert" panel. I'd been thinking a shift-click on the A#lock, which is what I'm trying in my Tamper script...

**Comment by majkaz** [comment]
Another note - not sure if this is only my setup - but Firefox (Firefox portable, v.50.0.2) downloads the gpx wrong - it opens it as gpx.html and saves it the same, using DownloadThemAll tools.

For mapper using JOSM it is sometimes useful to get the gpx of several squares, if a map item goes through them - if the validation check before uploading shows error or there is obvious mistake visible, sometimes the best way is to download complete secondary area and do the correction this way. An advantage is that if I download the gpx boundaries of several squares (small file download), I can often exactly pinpoint where the problem comes from, without downloading every map feature. And I can leave a comment in the relevant task as well. Sometimes, the problem isn't the next task (wouldn't need the gpx) but few tasks over...
